src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/RethrowToPreserveStackDetails.cs

```diff
@@ -5,10 +5,13 @@
 using Microsoft.CodeAnalysis;
 using Analyzer.Utilities;
 using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.CSharp;
```

Comment:
Why do you need `using Microsoft.CodeAnalysis.CSharp;`?

Reply:
Hmm interesting. Seems like it is not necessary but unnecessary using is not triggered for it.

Reply:
Oh awesome! I did reported an issue for this on roslyn but couldn't manage to create a reproducer I could share (it was on some internal company code).

Reply:
```suggestion
```

Reply:
the copmiler may be seeing a type in that and considering it during lookup while doing lambda overload resolution.  If you comment out your `context =>` lambda does this go away?

Reply:
if so, this is one of those cases where the namespace is marked by the compiler because it did examien it, but eventually ended up with a resolution taht did not use anything from it.  if so, this is a known limitation/extremely-difficult-problem in this space.